experimental/addons/crane/definitions/effective-hpa.cue

```diff
@@ -0,0 +1,80 @@
+"effective-hpa": {
+	alias: "ehpa"
+	annotations: {}
+	attributes: {
+		appliesToWorkloads: [
+			"deployments.apps",
+			"replicasets.apps",
+			"statefulsets.apps",
+		]
+		podDisruptive: false
+	}
+	description: "EffectiveHorizontalPodAutoscaler supports prediction-driven autoscaling and helps you manage application scaling in an easy way."
+	labels: {
+		"ui-hidden": "true"
+	}
+	type: "trait"
+}
+
+template: {
+	outputs: ehpa: {
+		apiVersion: "autoscaling.crane.io/v1alpha1"
+		kind:       "EffectiveHorizontalPodAutoscaler"
+		metadata: {
+			name:      context.appName + "-" + context.name
+			namespace: context.namespace
+		}
+		spec: {
+			scaleTargetRef: {
+				apiVersion: parameter.targetAPIVersion
+				kind:       parameter.targetKind
+				name:       context.name
```

Comment:
These info can be extract from context.output. Please refer to kruise rollout addon.

Reply:
good to know. refactored